### PR TITLE
DeviceTwin: Adding unit tests for types_helper.go

### PR DESCRIPTION
**Pull request by lidiyag**
This adds test cases for pkg/devicetwin/dttype/types_helper.go

**What type of PR is this?**
 /kind test

**What this PR does / why we need it**:
This PR adds test cases for types_helper.go.

### CODE REVIEW

File: edge/pkg/devicetwin/dttype/types_helper_test.go [hunk 1]
```diff
@@ -0,0 +1,776 @@
...
+				t.Errorf("UnmarshalMembershipDetail() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+// TestUnmarshalMembershipUpdate is function to test UnmarshalMembershipUpdate().
+func TestUnmarshalMembershipUpdate(t *testing.T) {
+	var memUpdate MembershipUpdate
+	bytesMemUpdate, _ := json.Marshal(memUpdate)
```
Comment (sids-b) [inline edge/pkg/devicetwin/dttype/types_helper_test.go]: handle error

File: edge/pkg/devicetwin/dttype/types_helper_test.go [hunk 2]
```diff
@@ -0,0 +1,776 @@
...
+				t.Errorf("UnmarshalMembershipUpdate() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+// TestUnmarshalBaseMessage is function to test UnmarshalBaseMessage().
+func TestUnmarshalBaseMessage(t *testing.T) {
+	var baseMessage BaseMessage
+	bytesBaseMessage, _ := json.Marshal(baseMessage)
```
Comment (sids-b) [inline edge/pkg/devicetwin/dttype/types_helper_test.go]: handle error

File: edge/pkg/devicetwin/dttype/types_helper_test.go [hunk 3]
```diff
@@ -0,0 +1,776 @@
...
+			if len(got) != len(tt.want) {
+				t.Errorf("DeviceAttrToMsgAttr failed due to wrong map size, Got Size = %v Want = %v", len(got), len(tt.want))
+			}
+			for gotKey, gotValue := range got {
+				keyPresent := false
+				for wantKey, wantValue := range tt.want {
+					if gotKey == wantKey {
+						keyPresent = true
+						if !reflect.DeepEqual(gotValue.Metadata, wantValue.Metadata) || !reflect.DeepEqual(gotValue.Optional, wantValue.Optional) || !reflect.DeepEqual(gotValue.Value, wantValue.Value) {
+							t.Errorf("Error in DeviceAttrToMsgAttr() Got wrong value for key %v", gotKey)
```
Comment (sids-b) [inline edge/pkg/devicetwin/dttype/types_helper_test.go]: add return ?

File: edge/pkg/devicetwin/dttype/types_helper_test.go [hunk 4]
```diff
@@ -0,0 +1,776 @@
...
+		})
+	}
+}
+
+//TestBuildDeviceState is function to test BuildDeviceState().
+func TestBuildDeviceState(t *testing.T) {
+	baseMessage := BaseMessage{EventID: uuid.NewV4().String(), Timestamp: time.Now().UnixNano() / 1e6}
+	device := Device{Name: "SensorTag", State: "ON", LastOnline: "Today"}
+	deviceMsg := DeviceMsg{BaseMessage: baseMessage, Device: device}
+	want, _ := json.Marshal(deviceMsg)
```
Comment (sids-b) [inline edge/pkg/devicetwin/dttype/types_helper_test.go]: handle error

File: edge/pkg/devicetwin/dttype/types_helper_test.go [hunk 5]
```diff
@@ -0,0 +1,776 @@
...
+}
+
+// TestBuildDeviceAttrUpdate is function to test BuildDeviceAttrUpdate().
+func TestBuildDeviceAttrUpdate(t *testing.T) {
+	baseMessage := BaseMessage{EventID: uuid.NewV4().String(), Timestamp: time.Now().UnixNano() / 1e6}
+	attr := dtclient.DeviceAttr{ID: 00, DeviceID: "DeviceA", Name: "SensorTag", Description: "Sensor", Value: "Temperature", Optional: true, AttrType: "float", Metadata: "CelsiusScale"}
+	attrs := make(map[string]*MsgAttr)
+	attrs[attr.Name] = &MsgAttr{Value: attr.Value, Optional: &attr.Optional, Metadata: &TypeMetadata{Type: attr.AttrType}}
+	devAttrUpdate := DeviceAttrUpdate{BaseMessage: baseMessage, Attributes: attrs}
+	bytesDevAttrUpdate, _ := json.Marshal(devAttrUpdate)
```
Comment (sids-b) [inline edge/pkg/devicetwin/dttype/types_helper_test.go]: handle error

File: edge/pkg/devicetwin/dttype/types_helper_test.go [hunk 6]
```diff
@@ -0,0 +1,776 @@
...
+}
+
+// TestBuildDeviceTwinResult is function to test BuildDeviceTwinResult().
+func TestBuildDeviceTwinResult(t *testing.T) {
+	baseMessage := BaseMessage{EventID: uuid.NewV4().String(), Timestamp: time.Now().UnixNano() / 1e6}
+	msgTwins := createMessageTwin()
+	devTwinResultDealType0 := createDeviceTwinResultDealType0(baseMessage)
+	bytesDealType0, _ := json.Marshal(devTwinResultDealType0)
+	devTwinResult1 := createDeviceTwinResult(baseMessage)
+	bytes_DealType1, _ := json.Marshal(devTwinResult1)
```
Comment (sids-b) [inline edge/pkg/devicetwin/dttype/types_helper_test.go]: handle error. applicable to all cases

File: edge/pkg/devicetwin/dttype/types_helper_test.go [hunk 7]
```diff
@@ -0,0 +1,776 @@
...
+	msgTwins := make(map[string]*MsgTwin)
+	twinMetadataDeleted := MsgTwin{Metadata: &TypeMetadata{Type: "deleted"}}
+	twinMetadataUpdated := MsgTwin{Metadata: &TypeMetadata{Type: "updated"}}
+	msgTwins["empty"] = nil
+	msgTwins["deleted"] = &twinMetadataDeleted
+	msgTwins["updated"] = &twinMetadataUpdated
+	return msgTwins
+}
+
+// createDeviceTwinResultDealType0() is function to create DeviceTwinResult with DealType 0.
```
Comment (rohitsardesai83) [inline edge/pkg/devicetwin/dttype/types_helper_test.go]: DealType0 -> what is type 0 ? can we have the proper type mentioned here?

Comment (lidiyag) [reply edge/pkg/devicetwin/dttype/types_helper_test.go]: DealType0 is Get...Updated

File: edge/pkg/devicetwin/dttype/types_helper_test.go [hunk 8]
```diff
@@ -0,0 +1,776 @@
...
+
+import (
+	"encoding/json"
+	"errors"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dtclient"
+	"github.com/satori/go.uuid"
```
[the remark this answers is not in the document]
Comment (lidiyag) [reply edge/pkg/devicetwin/dttype/types_helper_test.go]: Updated

File: edge/pkg/devicetwin/dttype/types_helper_test.go [hunk 9]
```diff
@@ -0,0 +1,776 @@
+package dttype
```
[the remark this answers is not in the document]
Comment (lidiyag) [reply edge/pkg/devicetwin/dttype/types_helper_test.go]: Updated

**Pull request merged**